src/query/interpret/eval.cpp

```diff
@@ -22,6 +22,16 @@ int64_t EvaluateInt(ExpressionEvaluator *evaluator, Expression *expr, std::strin
   }
 }
 
+std::optional<int64_t> EvaluateHopsLimit(ExpressionVisitor<TypedValue> &eval, Expression *expr) {
+  if (!expr) return std::nullopt;
+  TypedValue value = expr->Accept(eval);
+  try {
+    return value.ValueInt();
```

Comment:
you can also check if the value is non negative and make an exception there as well